File: backend/GameApp.Application/Services/SceneServices/SceneCreateService.cs
using GameApp.Application.UseCases.SceneUseCases;
using GameApp.Domain.Entities.Scenes;
using GameApp.Domain.Repositories;
namespace GameApp.Application.Services.SceneServices;

public class SceneCreateService : SceneCreateUseCase
{
    private readonly ISceneRepository _repo;

    public SceneCreateService(ISceneRepository repo) => _repo = repo;

    public async Task<Scene?> CreateScene(Scene scene)
    {
        // scene name must be unique in the collection
        var existingScene = await _repo.FetchByName(scene.GetName());
        if (existingScene is not null)
        {
            Console.WriteLine($"Scene name {scene.GetName()} exist");
            return null;
        }
        return await _repo.SaveAsync(scene);
    }
}

File: frontend/gameapp-frontend/src/application/services/scene-services/scene-create-service.ts
import type { ISceneCreateUseCase } from "../../usecases/scene-use-cases/scene-create-use-case";
import type { ISceneRepository } from "../../repositories/iscene-repository";
import { Scene } from "../../../domain/entities/scenes/scene";

export class SceneCreateService implements ISceneCreateUseCase {
    constructor(private sceneRepository: ISceneRepository) {}

    async createScene(scene: Scene): Promise<Scene> {
        const savedScene = await this.sceneRepository.save(scene);
        return savedScene;
    }
}
